src/main/java/bobby/Parser.java

```diff
@@ -0,0 +1,99 @@
+package bobby;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * The Parser class deals with converting user input into commands
+ * understood by the program and executes them accordingly.
+ */
+public class Parser {
+    public Parser() {
+
+    }
+
+    /**
+     * Converts user input into commands, passes the commands to TaskList
+     * to be executed, or terminates the Bobby program.
+     * @param tasks TaskList that handles the execution of commands.
+     * @param userInput user input to be parsed.
+     * @param bobby instance of Bobby to be terminated if BYE command given.
+     */
+    public static void parse(TaskList tasks, String userInput, Bobby bobby) {
+        String[] inputs = userInput.split(" ", 2);
+        String command = inputs[0];
+        try {
+            switch (Commands.valueOf(command.toUpperCase())) {
+            case BYE:
+                Ui.printExit();
+                bobby.terminate();
+                break;
+            case LIST:
+                tasks.list();
+                break;
+            case MARK:
+                try {
+                    int i = Integer.parseInt(inputs[1]) - 1;
+                    tasks.mark(i);
+                } catch (BobbyException e) {
+                    System.out.println(e);
+                }
+                break;
+            case UNMARK:
+                try {
+                    int k = Integer.parseInt(inputs[1]) - 1;
+                    tasks.unmark(k);
+                } catch (BobbyException e) {
+                    System.out.println(e);
+                }
+                break;
+            case TODO:
+                try {
+                    tasks.addToDo(userInput);
+                } catch (BobbyException e) {
+                    System.out.println(e);
+                }
+                break;
+            case DEADLINE:
+                try {
+                    tasks.addDeadline(userInput);
+                } catch (BobbyException e) {
```

Comment:
I think you can combine all the expected exceptions into one file and call that instance